api/marketdata.go

```diff
@@ -172,6 +200,17 @@ func getChartsHandler(charts *market.Charts) func(c *gin.Context) {
 			return
 		}
 
+		rawData, err := json.Marshal(chart)
+		if err == nil && !chart.IsEmpty() {
+			saveResult, err := cache.DB.Set(key, storage.CacheData{
+				RawData:      rawData,
+				WasSavedTime: timeStart,
+			})
+			if err != nil || saveResult != storage.SaveResultSuccess {
+				logger.Error("failed to save cache", logger.Params{"key": key, "coin": coinId, "currency": currency, "token": token, "time_start": timeStart})
+			}
+		}
+
```

Comment:
not tested also arguably shouldn't be in the controller but encapsulated elsewhere

Reply:
Is tested and tightly linked to the controller logic